core/tc.cc

```diff
@@ -606,62 +607,66 @@ static inline struct task_result tc_scheduled(struct tc *c) {
   return (struct task_result){.packets = 0, .bits = 0};
 }
 
-void sched_loop(struct sched *s) {
-  struct sched_stats last_stats = s->stats;
-  uint64_t last_print_tsc;
-  uint64_t checkpoint;
-  uint64_t now;
-
-  const double ns_per_cycle = 1e9 / tsc_hz;
-
-  last_print_tsc = checkpoint = now = rdtsc();
-
-  /* the main scheduling - running - accounting loop */
-  for (uint64_t round = 0;; round++) {
-    struct tc *c;
-    struct task_result ret;
-    resource_arr_t usage;
-
-    /* periodic check for every 2^8 rounds,
-     * to mitigate expensive operations */
-    if ((round & 0xff) == 0) {
-      if (unlikely(ctx.is_pause_requested())) {
-        if (unlikely(ctx.Block())) break;
-        last_stats = s->stats;
-        last_print_tsc = checkpoint = now = rdtsc();
-      } else if (unlikely(FLAGS_s && now - last_print_tsc >= tsc_hz)) {
-        print_stats(s, &last_stats);
-        last_stats = s->stats;
-        last_print_tsc = checkpoint = now = rdtsc();
+inline bool schedule_once(uint64_t round, struct sched *s) {
+  static struct sched_stats last_stats = s->stats;
```

Comment:
You should not use local-scoped static variables here, since all worker threads will contend on the same memory address.